batcher/aligned-batcher/src/connection.rs

```diff
@@ -21,7 +21,9 @@ pub(crate) async fn send_batch_inclusion_data_responses(
     finalized_batch: Vec<BatchQueueEntry>,
     batch_merkle_tree: &MerkleTree<VerificationCommitmentBatch>,
 ) -> Result<(), BatcherError> {
-    for (vd_batch_idx, entry) in finalized_batch.iter().enumerate() {
+    // iter in reverse because each sender wants to receive responses in ascending nonce order
+    // and finalized_batch is ordered as the PriorityQueue , low max_nonce first && high nonce first.
```

Comment:
nit: 
When we revert this vec, we are also reverting the orden of the proofs to low `max_fee` first, and higher `max_fee` last, too.
This shouldn't be a problem, since the sender is waiting until arrival of the last nonce, it doesn't care about the fees.
I think we should add a comment clarifying that.

Reply:
you are right, this is not good. because the low max_fee tends to be the high nonce, and we want the high nonce to be last.

Reply:
Priority queue orders first by max_fee.
This means the priorityQueue has ascending max_fee (we first eliminate those with low max_fee). And when max_fee is equal, it has descending nonce (we first eliminate those with high nonce)

This means priorityQueue.rev() has descending max_fee. And when the max_fee is equal, it will have ascending nonce. 

This is the desired behaviour, since descending max_fee => ascending nonce, and equal fee => ascending nonce.